pyro/distributions/dequantized.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+
+import torch
+from torch.distributions import constraints
+
+from .torch_distribution import TorchDistribution
+
+
+class DequantizedDistribution(TorchDistribution):
+    """
+    Dequantized distribution over relaxed counts.
+
+    The density of this distribution linearly interpolates the density of
+    ``base_dist``, except in the interval ``[0,1]`` where the weight of 0 is
+    doubled to account for asymmetry.
+
+    Given a distribution ``base_dist`` over nonnegative integers, this creates
+    a distribution over positive reals such that randomly rounded samples from
+    this distribution will be distributed approximately according to
+    ``base_dist``.
+
+    For example the following two models result in approximately identical
+    distributions over ``z``, conditioned on inputs ``n, p``.
+
+        # Model 1.
+        z ~ Binomial(n, p)
+
+        # Model 2.
+        r ~ DequantizedDistribution(Binomial(n, p))
+        b ~ Bernoulli(1 + floor(r) - r)      # Quantize.
```

Comment:
I think this is `r - floor(r)`, e.g. when r ~ 4.99, the Bernoulli prob should be ~ 0.99